deploy-agent/deployd/agent.py

```diff
@@ -455,7 +458,7 @@ def _update_deploy_alias(self, deploy_goal):
                                                                   deploy_goal.envName))
 
     @staticmethod
-    def plan_changed(old_response, new_response):
+    def plan_changed(old_response, new_response) -> Union[PingResponse, bool, DeployGoal]:
```

Comment:
this might be a signal that we need to refactor the code to have a single return type. Btw, can we use `PingResponse | bool | DeployGoal`?

Reply:
Agreed to your first point. That syntax is only supported in python 3.10 and above. https://peps.python.org/pep-0604/